tests/files_test.py

```diff
@@ -0,0 +1,262 @@
+        return_value="8cd45159712eeb914c049c717d3f4d75"
+    )
+    requests.post("http://" + mocket.MOCK_HOST_1 + "/post", files=file_data)
+
+    sock.connect.assert_called_once_with((mocket.MOCK_POOL_IP, 80))
+    sock.send.assert_has_calls(
+        [
+            mock.call(b"Content-Type"),
+            mock.call(b": "),
+            mock.call(
+                b"multipart/form-data; boundary=8cd45159712eeb914c049c717d3f4d75"
+            ),
+            mock.call(b"\r\n"),
+        ]
+    )
+    sock.send.assert_has_calls(
+        [
+            mock.call(b"Content-Length"),
+            mock.call(b": "),
+            mock.call(b"131"),
+            mock.call(b"\r\n"),
+        ]
+    )
+    sock.send.assert_has_calls(
+        [
+            mock.call(
+                b'--8cd45159712eeb914c049c717d3f4d75\r\nContent-Disposition: form-data; name="key_4"\r\n\r\n'
+            ),
+            mock.call(b"Value 5\r\n"),
+            mock.call(b"--8cd45159712eeb914c049c717d3f4d75--\r\n"),
+        ]
+    )
+
+
+def test_post_files_file(sock, requests):
+    with open("tests/files/red_green.png", "rb") as file_1:
+        file_data = {
+            "file_1": (
+                "red_green.png",
+                file_1,
+                "image/png",
+                {
+                    "Key_1": "Value 1",
+                    "Key_2": "Value 2",
+                    "Key_3": "Value 3",
+                },
+            ),
+        }
+
+        requests._build_boundary_string = mock.Mock(
+            return_value="e663061c5bfcc53139c8f68d016cbef3"
+        )
+        requests.post("http://" + mocket.MOCK_HOST_1 + "/post", files=file_data)
+
+    sock.connect.assert_called_once_with((mocket.MOCK_POOL_IP, 80))
+    sock.send.assert_has_calls(
+        [
+            mock.call(b"Content-Type"),
+            mock.call(b": "),
+            mock.call(
+                b"multipart/form-data; boundary=e663061c5bfcc53139c8f68d016cbef3"
+            ),
+            mock.call(b"\r\n"),
+        ]
+    )
+    sock.send.assert_has_calls(
+        [
+            mock.call(b"Content-Length"),
+            mock.call(b": "),
+            mock.call(b"347"),
+            mock.call(b"\r\n"),
+        ]
+    )
+    sock.send.assert_has_calls(
+        [
+            mock.call(
+                b'--e663061c5bfcc53139c8f68d016cbef3\r\nContent-Disposition: form-data; name="file_1"; filename="red_green.png"\r\nContent-Type: image/png\r\nKey_1: Value 1\r\nKey_2: Value 2\r\nKey_3: Value 3\r\n\r\n'
+            ),
+            mock.call(
+                b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x02\x00\x00\x00\x02\x08\x02\x00\x00\x00\xfd\xd4\x9a"
+            ),
+            mock.call(
+                b"s\x00\x00\x00\x01sRGB\x00\xae\xce\x1c\xe9\x00\x00\x00\x04gAMA\x00\x00\xb1\x8f\x0b\xfca\x05\x00\x00"
+            ),
+            mock.call(
+                b"\x00\tpHYs\x00\x00\x0e\xc3\x00\x00\x0e\xc3\x01\xc7o\xa8d\x00\x00\x00\x10IDAT\x18Wc\xf8\xcf"
```

Comment:
I don't really understand what these `call(b"big string of bytes")` are doing or what the bytes are and I'm a little hesitant about merging tests that pass weird bytestrings to `call()` methods without having a better understanding of what they are or what it does functionally. 

I guess maybe the PNG data based on that appearing in the first one, but I'm not sure. If that is the case I'm in favor of maybe using PIL or something that is comparing image file to image file ideally with less need for the "magic" bytestrings hardcoded into the test.

You can submit these back as a PR to the main repo if you want.